### PR TITLE
Separate build step and test step in GitHub Actions build matrix

### PR DESCRIPTION
**Pull request by BrettDong**
#### Summary
None

### CODE REVIEW

**Comment by kevingranade** [comment]
Better readability, more flexible, wins all around.

File: .github/workflows/matrix.yml [hunk 1]
```diff
@@ -243,9 +243,28 @@ jobs:
...
+      run: |
+        find . -type f -name "*cache-*" -exec tar -rvf ccache.debug.tar {} \;
+    - name: upload ccache debug files
+      uses: actions/upload-artifact@v2
+      with:
+        name: ccache-debug
+        path: ccache.debug.tar
+        if-no-files-found: ignore
+    - name: cleanup ccache before upload
+      if: ${{ env.skip == 'false' && !failure() && (runner.os == 'Linux' || runner.os == 'macOS') }}
```
Comment (akrieger) [inline .github/workflows/matrix.yml:259]:
```suggestion
      if: ${{ env.SKIP == 'false' && !failure() && (runner.os == 'Linux' || runner.os == 'macOS') }}
```

Comment (akrieger) [reply .github/workflows/matrix.yml:259]: You'll delete the original at the bottom if you move this up here?

Comment (BrettDong) [reply .github/workflows/matrix.yml:259]: Yes I forget to delete the old one.

**Pull request merged**